src/main/java/igym/exceptions/GlobalExceptionHandler.java

```
@@ -42,6 +42,18 @@ public ResponseEntity<Object> handleDuplicateGymException(DuplicateGymException
         return ResponseEntity.status(HttpStatus.CONFLICT).body(body);
     }
 
+    @ExceptionHandler(GymNotFoundException.class)
+    public ResponseEntity<Object> handleGymNotFoundException(GymNotFoundException ex) {
+        Map<String, Object> body = buildResponseBody(HttpStatus.NOT_FOUND, ex.getMessage());
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(body);
+    }
+
+    @ExceptionHandler(GymAlreadyDeletedException.class)
```

Daniel, I think we do not need to handle this exception this way.
Instead, we can just check if the Gym exists AND is active. If the gym doesn't exist or it exists but is inactive, so we can throw the 404 error. 
Remember, internally, we are making a "soft deletion", but to external users of the API, it appears as a "hard deletion". So, if it was deleted, it can't return a "conflict".

---

Nice, I got it, thank you for explanation